src/read_write/raw.rs

```diff
@@ -107,45 +109,228 @@ impl NodeReader for RawNodeReader {
             }
         }
 
-        point.color.red = self.attribute_readers[0].read_u8()?;
-        point.color.green = self.attribute_readers[0].read_u8()?;
-        point.color.blue = self.attribute_readers[0].read_u8()?;
+        if let Some(cr) = self.attribute_readers.get_mut("color") {
+            point.color.red = cr.reader.read_u8()?;
+            point.color.green = cr.reader.read_u8()?;
+            point.color.blue = cr.reader.read_u8()?;
+        }
 
-        if let Some(ir) = self.attribute_readers.get_mut(1) {
-            point.intensity = Some(ir.read_f32::<LittleEndian>()?);
+        if let Some(ir) = self.attribute_readers.get_mut("intensity") {
+            point.intensity = Some(ir.reader.read_f32::<LittleEndian>()?);
         }
 
         Ok(point)
     }
 }
 
+impl RawNodeReader {
+    #[allow(dead_code)]
+    fn read_batch(&mut self) -> Option<PointsBatch> {
+        let mut batch = PointsBatch {
+            position: vec![],
+            attributes: BTreeMap::new(),
+        };
+
+        let _err: io::Result<()> = match self.encoding {
+            Encoding::Plain => (0..NUM_POINTS_PER_BATCH).try_for_each(|_| {
+                let x = self.xyz_reader.read_f64::<LittleEndian>()?;
+                let y = self.xyz_reader.read_f64::<LittleEndian>()?;
+                let z = self.xyz_reader.read_f64::<LittleEndian>()?;
+                batch.position.push(Vector3::new(x, y, z));
+                Ok(())
+            }),
+            Encoding::ScaledToCube(min, edge_length, PositionEncoding::Uint8) => {
```

Comment:
Similar to above, how about an inner `match` against `position_encoding`?

Reply:
Done.